crates/evm/core/src/backend/strategy.rs

```diff
@@ -181,6 +196,48 @@ impl BackendStrategyRunner for EvmBackendStrategyRunner {
     ) {
         EvmBackendMergeStrategy::merge_db_account_data(addr, active, fork_db);
     }
+
+    // Note(zk): This function in upstream code is not implemented as part of the strategy pattern,
+    // but is instead a standalone function. We have moved it here to make it part of the
+    // strategy, as there is some abstraction in the middle since the envelopes, and types are
+    // different. The changes are:
+    // - The function signature has been changed to take a `Bytes` instead of `the
+    //   TransactionRequest`
+    // - The function signature has been changed to return a `TransactionMaybeSigned` instead of
+    //   empty tuple
+    // - Avoids some cloning of the backend state
+    // See the main function in entrypoint: crates/evm/core/src/backend/mod.rs
+    // See zk implementation in: crates/strategy/zksync/src/backend/runner.rs
+    fn transact_from_tx(
+        &self,
+        backend: &mut Backend,
+        data: &Bytes,
+        mut env: Env,
+        journaled_state: &mut JournaledState,
+        inspector: &mut dyn InspectorExt,
+    ) -> eyre::Result<TransactionMaybeSigned> {
+        let envelope: TxEnvelope = TxEnvelope::decode(&mut data.as_ref())
+            .map_err(|err| eyre::eyre!("failed to decode RLP-encoded transaction: {err}"))?;
+
+        let tx: &TransactionRequest = &envelope.clone().into();
```

Comment:
This is fine, but I think it's uncommon to `.into()` in a reference directly, and instead it's preferred to go with `&tx` later on when needed :) I believe it would also let you drop the explicit type.

Take is as a nit to learn more idiomatic rust :)